### ISSUE TITLE
Batch size changes for validation set during training?

### ISSUE THREAD

**Issue by Garakhan** [ultralytics/yolov5#11276]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

![Screenshot from 2023-03-30 16-49-28](https://user-images.githubusercontent.com/54187927/228841305-64b9b195-9719-45cd-966c-af4eb2575df3.png)

I am training yolov5s with custom dataset of batch size 12. The iterations for training set is OK when calculating `training set / batch size` (division). Bu the iterations for validations set, which is 67, does not work out with `val set / batch size`. Val set is 1642, so the iteration should be around 25.  Does it mean that batch size for validation set is different than 12?

**Comment by glenn-jocher** [comment]
@Garakhan val batch size is set to be double the train batch size. This is done for faster val, and since val takes less CUDA memory per image since no gradient is computed.

**Comment by Garakhan** [comment]
@glenn-jocher Thank you for the explanation.

**Comment by halqadasi** [comment]
> @Garakhan val batch size is set to be double the train batch size. This is done for faster val, and since val takes less CUDA memory per image since no gradient is computed.

Grateful for this info. I was confused so much about that. Finally, I found it.
